pulp_ansible/app/tasks/collections.py

```diff
@@ -81,6 +82,32 @@ def sync(remote_pk, repository_pk, mirror):
     d_version = AnsibleDeclarativeVersion(first_stage, repository, mirror=mirror)
     d_version.create()
 
+    repo_version = repository.latest_version()
+    to_deprecate = []
+    to_deprecate_query = Q()
+    to_undeprecate_query = Q()
+    for collection, deprecated in first_stage.deprecations.items():
```

Comment:
I see one way this could be even higher performing without much effort. Try having deprecations contain the pre-built `Q()` objects, and only store those where `deprecated=True`. This improves several things:

1. It doesn't have to re-loop through every collection here, just those that are deprecated. The number of `deprecated = True` will likely be few.
2. It doesn't have to join them with a '.' only to resplit them again later